Makefile

```diff
@@ -202,13 +202,15 @@ PROMTAIL_GO_FLAGS := $(GO_FLAGS)
 PROMTAIL_DEBUG_GO_FLAGS := $(DEBUG_GO_FLAGS)
 
 # Validate GOHOSTOS=linux && GOOS=linux to use CGO.
+ifneq ($(CGO_ENABLED), 0)
```

Comment:
Thanks for the PR :)

I like it is being controlled by `CGO_ENABLED`. But explained my concern here. https://github.com/grafana/loki/issues/3933#issuecomment-872848303

why not just `PROMTAIL_CGO = $(CGO_ENABLED)` or something similar?

Reply:
I've reworked the MR to only affect PROMTAIL_CGO. In my previous tests, it didn't work if I only modified that, but that seems to have been a problem with that specific test, it now works fine.